XenAdmin/Commands/SuspendVMCommand.cs

```diff
@@ -201,10 +201,18 @@ protected override string GetCantExecuteReasonCore(SelectedItem item)
             {
                 return Messages.VM_SHUT_DOWN;
             }
-            else if (vm.power_state == vm_power_state.Suspended)
+            if (vm.power_state == vm_power_state.Suspended)
             {
                 return Messages.VM_ALREADY_SUSPENDED;
             }
+            if (vm.HasGPUPassthrough())
+            {
+                return FriendlyErrorNames.VM_HAS_PCI_ATTACHED;

```

Comment:
FriendlyErrorNames.VM_HAS_PCI_ATTACHED says "This operation could not be performed..." and FriendlyErrorNames.VM_LACKS_FEATURE_SUSPEND says "You attempted an operation...". These messages are not suitable for a **tooltip**, because at that point no operation has been attempted.

Reply:
You suggested yesterday using VM.assert_operation_valid, would this be what I should be aiming for here as well?

Reply:
As per our discussion this is resolved for pool-admin and higher user roles using the API call VM.assert_operation_valid without needing to independently figure out what the error could be.

However it will revert back to "Unknown" for roles vm-operator (lowest that is allowed to Suspend) up to but not including pool-admin (assert_operation_valid minimum).

Reply:
I think that we shouldn't use the assert_operation_valid api call if it does not work for all users (although fixed in master, XenCenter needs a solution for older hosts as well). It's a shame, because it was a good idea. 
So I think we need to go back to the initial idea and capture the two cases (GPU passthrough and vGPU) and present specific "reasons" for those two.

Reply:
I have committed for discussion my current best estimate of the changes required. I may need assistance in replicating the issue.

Reply:
I have been able to overcome the problems I was having replicating the issue, this would appear to me to be now a workable solution.